webapp/client/src/components/FormFieldRadio.js

```diff
@@ -0,0 +1,132 @@
+import { useState } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import FormFieldScaffolding from "./FormFieldScaffolding";
+import FieldLabelForSeparatedFields from "./FieldLabelForSeparatedFields";
+import { optionsPropType } from "../lib/propTypes";
+
+const Radio = styled.div`
+  input[type="radio"] {
+    opacity: 0;
+    position: absolute;
+  }
+
+  input[type="radio"] + label {
+    display: flex;
+    padding-left: 0;
+    padding-top: 0;
+    font-size: var(--text-medium);
+  }
+
+  input[type="radio"] + label::before {
+    content: "";
+    display: inline-block;
+    vertical-align: bottom;
+    width: 30px;
+    height: 30px;
+    min-width: 30px;
+    margin-right: 12px;
+    background: url("/icons/radio_button_default.svg") no-repeat center;
+  }
+
+  input[type="radio"]:checked + label::before {
+    background: url("/icons/radio_button_checked.svg") no-repeat center;
+  }
+
+  input[type="radio"]:checked + label:hover::before {
+    background: url("/icons/radio_button_checked_hover.svg") no-repeat center;
+  }
+
+  input[type="radio"]:not(:checked):focus + label::before {
+    background: url("/icons/radio_button_focus.svg") no-repeat center;
+  }
+
+  input[type="radio"]:checked:focus + label::before {
+    background: url("/icons/radio_button_checked_focus.svg") no-repeat center;
+  }
+
+  .radio-button-list {
+    padding-top: var(--spacing-02);
+  }
+`;
+
+function FormFieldRadio({
+  fieldName,
+  fieldId,
+  options,
+  value,
+  required,
+  autofocus,
+  label,
+  details,
+  errors,
+  onChangeHandler,
+}) {
+  const [selectedValue, setSelectedValue] = useState(value);
+
+  const toggleRadioButton = (event) => {
+    setSelectedValue(event.target.value);
+  };
+
+  return (
+    <FormFieldScaffolding
+      {...{
+        fieldName,
+        errors,
+      }}
+      hideLabel
+      render={() => (
+        <Radio>
+          <fieldset id={fieldId} name={fieldId}>
+            <FieldLabelForSeparatedFields {...{ label, fieldId, details }} />
+            <div className="radio-button-list">
+              {options.map((option, i) => [
+                <input
+                  type="radio"
+                  id={fieldId + option.value}
+                  key={`${fieldId}-${option.value}`}
+                  name={fieldId}
+                  required={required}
+                  autoFocus={autofocus && i === 0}
+                  value={option.value}
+                  defaultChecked={selectedValue === option.value}
+                  onClick={toggleRadioButton}
```

Comment:
Have you tested keyboard navigation with this? Will `onClick` work when you tab and arrow-key your way to a radio button and hit space bar to select it? Just looking at the code, I'd expect this not to work but I could be wrong.

Reply:
I'm now seeing the automated tests, but worth confirming manually…

Reply:
Alright, I tried it out and it works 👍